static/js/components/PublishDrawer.tsx

```diff
@@ -123,7 +123,7 @@ const PublishingOption: React.FC<PublishingOptionProps> = (props) => {
           )}
           <PublishForm
             onSubmit={handlePublish}
-            disabled={!publishingInfo.hasUnpublishedChanges}
+            disabled={!publishingInfo.hasUnpublishedChanges || (!website.has_site_metadata && publishingEnv==="production")}
```

Comment:
This condition is rather long, maybe we could refactor it into a separate variable?

This is just a suggestion. It is up to your discretion.

```js
const isPublishDisabled = (
  !publishingInfo.hasUnpublishedChanges
  || !website.has_site_metadata && publishingEnv==="production"
)
```